puppy.py

```diff
@@ -28,4 +29,177 @@ def __init__(self, behavior_matches):
+        puppy.left_leg_motor.run_angle(speed, -40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, -40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, 40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, -40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, 40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, 40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, -40, wait=True)
+
+    def sit(left_leg_motor, right_leg_motor):
+        left_leg_motor.run(-1000)
+        right_leg_motor.run(-1000)
+        wait(2000)
+        left_leg_motor.stop()
+        right_leg_motor.stop()
+
+    def hop(left_leg_motor, right_leg_motor):
+        left_leg_motor.run(500)
+        right_leg_motor.run(500)
+        wait(275)
+        left_leg_motor.hold()
+        right_leg_motor.hold()
+        wait(275)
+        left_leg_motor.run(-50)
+        right_leg_motor.run(-50)
+        wait(275)
+        left_leg_motor.stop()
+        right_leg_motor.stop()
+
+    def step_0_node_0_action(self): 
+        ev3.screen.load_image(NEUTRAL_EYES)
+
+    def step_0_node_1_action(self):
+        if self.behavior_matches:
+            head_motor.run(1000)
+            wait(1000)
+            self.ev3.speaker.say("uh oh")
+            head_motor.run(-1000)
+            wait(1000)
+        else:
+            pass
+  
+    def step_1_node_0_action(self):
+        if self.behavior_matches:
+            walk_in_place(self, 100)
+        else:
+            pass
+    
+    def step_2_node_1_action(self): #vents
+        if self.behavior_matches:
+            head_motor.run(1000)
+            wait(10)
+            self.ev3.speaker.say("uh oh")
+            head_motor.run(-1000)
+            wait(10)
+            head_motor.stop()
+        else: 
+            pass
+    
+    def step_2_node_2_action(self): #helicopter
+        if self.behavior_matches:
+            self.ev3.light.on(Color.GREEN)
+            self.ev3.speaker.say("this way")
+            self.ev3.light.on(Color.ORANGE)
+            self.ev3.speaker.say("or that way")
+        else:
+            pass
+            
+    def step_3_node_1_action(self):
+        if self.behavior_matches:
+            sit(left_leg_motor, right_leg_motor)
+        else:
+            pass        
+    
+    def step_3_node_2_action(self):
+        if self.behavior_matches:
+            ev3.screen.load_image(TIRED_LEFT_EYES)
+            ev3.screen.load_image(TIRED_RIGHT_EYES)
+        else:
+            pass    
+    
+    def step_3_node_3_action(self):
+        if self.behavior_matches:
+            ev3.screen.load_image(SQUINTY_EYES)   
+            ev3.screen.load_image(HURT_EYES)   
+            head_motor.run(10)
+            wait(5)
+            head_motor.stop()
+        else:
+            pass    
 
+    def step_4_node_1_action(self):
+        if self.behavior_matches:
+            hop(left_leg_motor, right_leg_motor)
```

Comment:
similar to comment about sit, this should be self.hop(self.left_leg_motor, self.right_leg_motor)